app/src/main/java/io/intrepid/contest/screens/adminstatus/AdminStatusFragment.java

```diff
@@ -0,0 +1,87 @@
+package io.intrepid.contest.screens.adminstatus;
+
+import android.graphics.drawable.Drawable;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.widget.Button;
+import android.widget.TextView;
+
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.OnClick;
+import io.intrepid.contest.R;
+import io.intrepid.contest.base.BaseFragment;
+import io.intrepid.contest.base.PresenterConfiguration;
+import io.intrepid.contest.models.Entry;
+
+public class AdminStatusFragment extends BaseFragment<AdminStatusPresenter> implements AdminStatusContract.View {
+    @BindView(R.id.entry_review_recycler_view)
+    RecyclerView entriesRecyclerView;
+    @BindView(R.id.awaiting_submission_text_indicator)
+    TextView awaitingSubmissionsTextView;
+    @BindView(R.id.judging_text_indicator)
+    TextView judgingTextIndicator;
+    @BindView(R.id.end_of_contest_text_indicator)
+    TextView endOfContestTextIndicator;
+    @BindView(R.id.start_contest_button)
+    Button bottomNavigationButton;
+
+    private EntryReviewAdapter adapter = new EntryReviewAdapter();
+
+    @NonNull
+    @Override
+    public AdminStatusPresenter createPresenter(PresenterConfiguration configuration) {
+        return new AdminStatusPresenter(this, configuration);
+    }
+
+    @Override
+    protected int getLayoutResourceId() {
+        return R.layout.admin_status_fragment;
+    }
+
+    @OnClick(R.id.start_contest_button)
+    public void onStartButtonClicked() {
+        presenter.onStartButtonClicked();
+    }
+
+    @Override
+    protected void onViewCreated(@Nullable Bundle savedInstanceState) {
+        super.onViewCreated(savedInstanceState);
+        setActionBarTitle(R.string.contest_status_bar_title);
+        setActionBarDisplayHomeAsUpEnabled(true);
+        entriesRecyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
+        entriesRecyclerView.setAdapter(adapter);
+    }
+
+    @Override
+    public void showAwaitSubmissionsIndicator(boolean inProgress) {
+        int drawable = inProgress ? R.drawable.processing : R.drawable.check_with_circle;
```

Comment:
So we repeat essentially the same code in three places, let's extract these two lines into a private function that takes in the boolean inProgress, and a textview, then each of these indicator functions can just call that new function, passing in the inProgress boolean and the textview.